### PR TITLE
Extend START, END, READY controls to allow BOOL type

### PR DESCRIPTION
**Pull request by krishung5**

### CODE REVIEW

File: protobuf/model_config.proto [hunk 1]
```diff
@@ -255,7 +255,8 @@ message ModelInstanceGroup
 
   //@@  .. cpp:var:: SecondaryDevice secondary_devices (repeated)
   //@@
-  //@@     Secondary devices that are required by instances specified by this instance group. Optional.
+  //@@     Secondary devices that are required by instances specified by this
+  // instance group. Optional.
```
Comment (tanmayv25) [inline protobuf/model_config.proto:259]: //@@ instance group. Optional.

File: protobuf/model_config.proto [hunk 2]
```diff
@@ -1221,11 +1223,21 @@ message ModelSequenceBatching
...
+    //@@       the request. 'bool_false_true' must have two entries: the
+    //@@       first the false value and the second the true value.
+    //@@
+    repeated float bool_false_true = 4;
+
     //@@    .. cpp:var:: DataType data_type
     //@@
     //@@       The control's datatype.
     //@@
-    DataType data_type = 4;
```
Comment (tanmayv25) [inline protobuf/model_config.proto:1228]: For backwards compatibility for the protobuf files should not change the identifier of the fields. Leave `data_type = 4` and make `bool_false_true = 5`

> you must not change the tag numbers of any existing fields.

Source: https://developers.google.com/protocol-buffers/docs/pythontutorial#extending-a-protocol-buffer

File: protobuf/model_config.proto [hunk 3]
```diff
@@ -1221,11 +1223,21 @@ message ModelSequenceBatching
...
 
+    //@@    .. cpp:var:: bool bool_false_true (repeated)
+    //@@
+    //@@       The control's true and false setting is indicated by setting
+    //@@       a value in a bool tensor. The tensor must be a
+    //@@       1-dimensional tensor with size equal to the batch size of
+    //@@       the request. 'bool_false_true' must have two entries: the
+    //@@       first the false value and the second the true value.
+    //@@
+    repeated float bool_false_true = 4;
```
Comment (tanmayv25) [inline protobuf/model_config.proto:1234]: Shouldn't it be bool instead of float?

**Review by tanmayv25 — CHANGES_REQUESTED** [review]

File: protobuf/model_config.proto (hunk 2 above)
Comment (krishung5) [reply protobuf/model_config.proto:1228]: Thanks for the link. Updated.

File: protobuf/model_config.proto (hunk 3 above)
Comment (CoderHam) [reply protobuf/model_config.proto]: +1

**Review by CoderHam — APPROVED** [review]

File: protobuf/model_config.proto [hunk 4]
```diff
@@ -1226,6 +1228,16 @@ message ModelSequenceBatching
...
+
+    //@@    .. cpp:var:: bool bool_false_true (repeated)
+    //@@
+    //@@       The control's true and false setting is indicated by setting
+    //@@       a value in a bool tensor. The tensor must be a
+    //@@       1-dimensional tensor with size equal to the batch size of
+    //@@       the request. 'bool_false_true' must have two entries: the
+    //@@       first the false value and the second the true value.
+    //@@
+    repeated bool bool_false_true = 5;
```
Comment (deadeyegoodwin) [inline protobuf/model_config.proto:1240]: Put this after fp32_false_true

**Review by deadeyegoodwin — CHANGES_REQUESTED** [review]

File: protobuf/model_config.proto (hunk 4 above)
Comment (krishung5) [reply protobuf/model_config.proto:1234]: Updated

**Review by CoderHam — APPROVED** [review]

**Review by deadeyegoodwin — DISMISSED** [review]

**Review by tanmayv25 — APPROVED** [review]

**Pull request merged**